lib/src/layer/marker_layer.dart

```diff
@@ -1,271 +1,148 @@
-  /// [Directionality.of] returns	 [TextDirection.ltr], and `-1.0` if
-  /// [Directionality.of] returns [TextDirection.rtl].
+  /// When `true`, markers will always appear upright and vertical from the
+  /// user's perspective. Defaults to `false`. Overriden by [Marker.rotate].
   ///
-  /// Overriden on a per [Marker] basis if [Marker.rotateAlignment] is specified.
-  final AlignmentGeometry? rotateAlignment;
+  /// Note that this is not used to apply a custom rotation in degrees to the
+  /// markers. Use a widget inside [Marker.builder] to perform this.
+  final bool rotate;
 
   const MarkerLayer({
     super.key,
     this.markers = const [],
-    this.position,
+    this.alignment = Alignment.center,
     this.rotate = false,
-    this.rotateOrigin,
-    this.rotateAlignment,
   });
 
   @override
   Widget build(BuildContext context) {
     final map = MapCamera.of(context);
-    final markerWidgets = <Widget>[];
-
-    for (final marker in markers) {
-      final pxPoint = map.project(marker.point);
-
-      // See if any portion of the Marker rect resides in the map bounds
-      // If not, don't spend any resources on build function.
-      // This calculation works for any Anchor position whithin the Marker
-      // Note that Anchor coordinates of (0,0) are at bottom-right of the Marker
-      // unlike the map coordinates.
-      final anchor = marker.anchor ??
-          Anchor.fromPos(
-            position ?? AnchorPos.defaultAnchorPos,
-            marker.width,
-            marker.height,
-          );
-      final rightPortion = marker.width - anchor.left;
-      final leftPortion = anchor.left;
-      final bottomPortion = marker.height - anchor.top;
-      final topPortion = anchor.top;
-      if (!map.pixelBounds.containsPartialBounds(Bounds(
-          Point(pxPoint.x + leftPortion, pxPoint.y - bottomPortion),
-          Point(pxPoint.x - rightPortion, pxPoint.y + topPortion)))) {
-        continue;
-      }
 
-      final defaultRotateAlignment =
-          position != null ? position!.geometry * -1 : Alignment.center;
-
-      final pos = pxPoint.subtract(map.pixelOrigin);
-      final markerWidget = (marker.rotate ?? rotate)
-          ? Transform.rotate(
-              angle: -map.rotationRad,
-              origin: marker.rotateOrigin ?? rotateOrigin ?? Offset.zero,
-              alignment: marker.rotateAlignment ??
-                  rotateAlignment ??
-                  defaultRotateAlignment,
-              child: marker.builder(context),
-            )
-          : marker.builder(context);
-
-      markerWidgets.add(
-        Positioned(
-          key: marker.key,
-          width: marker.width,
-          height: marker.height,
-          left: pos.x - rightPortion,
-          top: pos.y - bottomPortion,
-          child: markerWidget,
-        ),
-      );
-    }
-    return Stack(children: markerWidgets);
+    return Stack(
+      children: markers
```

Comment:
In the spirit of micro-optimizations:

```dart
List<Widget>.generate(markers.length, (i) {
  final m = markers[i];

  // ...
})..retainWhere((w) => w != null)
```

is ~50% cheaper depending on the size and sparseness of the nullable iterable. I was concerned that the memory utilization might be higher by allocating a bigger chunk and then scratching elements out but against my intuition my micro-benchmark (don't trust micro benchmarks) showed the opposite. Explaining that would probably requiring digging deeper into the iterator impl.

Reply:
Yep, did consider the cheapest way of doing this and forgot that `generate` is insanely cheap for some reason. I'll change it.

(I can confirm with a bigger benchmark: FMTC. It drastically improved some throughput when used in a more complex situation.)

Reply:
>  for some reason

The difference is that generate knows the size so it can just pre-allocate an appropriate chunk of memory, whereas the map().filter().toList() approach builds an iterator and then pushes back elements, which leads to a series of iteratively larger allocations and repeated memcopies from the old to the new allocation. Thus, I'd expect the benefit to invert for very large, very sparse lists.

Reply:
@ignatz That code doesn't work by itself, as it requires `cast`ing the `List<Widget?>` to `List<Widget>` (`retainWhere` doesn't change the typing like `whereNotNull`). Could it still be faster even with a call to `cast()`?

Reply:
Ooops :). My expectation would be: yes. cast() doesn't return a new list (i.e new allocation) but rather a view wrapping the original list, which merely does the Widget <=> Widget? casting for you (See https://api.flutter.dev/flutter/dart-core/List/cast.html). I would still expect repeat allocations to be more expensive and to create more GC pressure than a vptr lookup. 

At the end of the day either approach will be fine.